### PR TITLE
Make preliminary adjustments to WaypointUpdater

### PR DESCRIPTION
**Pull request by mithi**
@mauricelos, @ljanyst, @buffos 

Sorry this took this long, apparently I allotted too small disk space for Ubuntu
(I allotted just 10GB because my mac only has 128 GB), so I had to free up a lot of space for this project.

I ran `catkin_make` and `pylint` just fine

When I made and ran `styx` and the simulator, there were no errors but
no info was printed in the terminal e.g. 
`rospy.loginfo('WaypointUpdater Initialized.')`

### CODE REVIEW

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 1]
```diff
@@ -31,131 +29,136 @@
...
+        #TODO: should we use something similar to distance() instead of get_square_gap() instead?
+
+        if gap < best_gap:
+            best_index, best_gap = i, gap
+
+    return best_index
+
+
+def get_next_waypoints_wrapped(waypoints, i, n):
+    """Returns a list of n waypoints ahead of the vehicle"""
```
Comment (ljanyst) [inline ros/src/waypoint_updater/waypoint_updater.py]: Example comment :P

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 2]
```diff
@@ -31,131 +29,136 @@
...
+def get_closest_waypoint_index(my_position, waypoints):
     """
-    STUB
+    position: geometry_msg.msgs.Pose instance
+    waypoints: list of styx_msgs.msg.Waypoint instances
+    returns index of the closest waypoint in the list waypoints
     """
-    closestLen = 100000.0
-    closestWaypoint = 0
+    best_gap = 1000000000000000000000000000000.0 # Really big number
```
Comment (ljanyst) [inline ros/src/waypoint_updater/waypoint_updater.py]: I think the proper idiom for this is `float('inf')`.

**Review by ljanyst — COMMENTED** [review]
Test

File: ros/src/waypoint_updater/waypoint_updater.py (hunk 2 above)
Comment (buffos) [reply ros/src/waypoint_updater/waypoint_updater.py]: Just testing

Comment (mithi) [reply ros/src/waypoint_updater/waypoint_updater.py]: I see it!

Comment (ljanyst) [reply ros/src/waypoint_updater/waypoint_updater.py]: Cool! :)

**Comment by ljanyst** [comment]
I see your log messages just fine.

screenshot_20170824_232724

**Comment by ljanyst** [comment]
Unless others have comments, I am fine with approving this PR.

**Review by ljanyst — CHANGES_REQUESTED** [review]
Please remove `ros/Luca Massaron.desktop` and `ros/src/CMakeLists.txt`. As a rule never do `git add -a`.

**Comment by mithi** [comment]
Okay so sorry about this! :(

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 3]
```diff
@@ -113,8 +113,9 @@ def __init__(self):
         # Current list of base waypoints from Lane object
         self.base_waypoints = None
 
-        rospy.loginfo('WaypointUpdater Initialized.')
         rospy.spin()
+        rospy.loginfo('WaypointUpdater Initialized.')
+        print "...Waypoints initalized"
```
Comment (ljanyst) [inline ros/src/waypoint_updater/waypoint_updater.py]: The two lines above will only be called after ROS has been terminated. It does not do what you expect it to do.

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 4]
```diff
@@ -98,8 +98,8 @@ class WaypointUpdater(object):
        points with target velocities representing path ahead"""
 
     def __init__(self):
+        print "initializing waypoints..."
```
Comment (ljanyst) [inline ros/src/waypoint_updater/waypoint_updater.py]: Please remove this.

**Review by ljanyst — APPROVED** [review]

**Comment by mithi** [comment]
I found him! I must have accidentally dragged the name to the folder! I'm so sorry!
Note to self: Never use `git add .` 
i_found_him

File: ros/src/waypoint_updater/waypoint_updater.py (hunk 3 above)
Comment (mithi) [reply ros/src/waypoint_updater/waypoint_updater.py]: My bad, it should be before the `spin()`

**Pull request merged**